pkg/apis/gcp/validation/backupbucket_test.go

```diff
@@ -9,77 +9,210 @@ import (
+				"Type":   Equal(field.ErrorTypeNotSupported),
+				"Field":  Equal("spec.credentialsRef"),
+				"Detail": Equal("supported values: \"/v1, Kind=Secret\", \"security.gardener.cloud/v1alpha1, Kind=WorkloadIdentity\""),
+			}))))
+		})
+
+		It("should allow v1.Secret credentials", func() {
+			credsRef := &corev1.ObjectReference{
+				APIVersion: "v1",
+				Kind:       "Secret",
+				Name:       "my-creds",
+				Namespace:  "my-namespace",
+			}
+			errs := ValidateBackupBucketCredentialsRef(credsRef, fldPath)
+			Expect(errs).To(BeEmpty())
+		})
+
+		It("should allow security.gardener.cloud/v1alpha1.WorkloadIdentity credentials", func() {
+			credsRef := &corev1.ObjectReference{
+				APIVersion: "security.gardener.cloud/v1alpha1",
+				Kind:       "WorkloadIdentity",
+				Name:       "my-creds",
+				Namespace:  "my-namespace",
 			}
-		},
-		Entry("valid config",
-			&apisgcp.BackupBucketConfig{
-				Immutability: &apisgcp.ImmutableConfig{
-					RetentionType:   "bucket",
-					RetentionPeriod: metav1.Duration{Duration: 24 * time.Hour},
-				},
-			}, false, ""),
-		Entry("missing retentionType",
-			&apisgcp.BackupBucketConfig{
-				Immutability: &apisgcp.ImmutableConfig{
-					RetentionType:   "",
-					RetentionPeriod: metav1.Duration{Duration: 1 * time.Hour},
-				},
-			}, true, "must be 'bucket'"),
-		Entry("invalid retentionType",
-			&apisgcp.BackupBucketConfig{
-				Immutability: &apisgcp.ImmutableConfig{
-					RetentionType:   "invalid",
-					RetentionPeriod: metav1.Duration{Duration: 1 * time.Hour},
-				},
-			}, true, "must be 'bucket'"),
-		Entry("non-positive retentionPeriod",
-			&apisgcp.BackupBucketConfig{
-				Immutability: &apisgcp.ImmutableConfig{
-					RetentionType:   "bucket",
-					RetentionPeriod: metav1.Duration{Duration: 0},
-				},
-			}, true, "must be a positive duration greater than 24h"),
-		Entry("negative retentionPeriod",
-			&apisgcp.BackupBucketConfig{
-				Immutability: &apisgcp.ImmutableConfig{
-					RetentionType:   "bucket",
-					RetentionPeriod: metav1.Duration{Duration: -1 * time.Hour},
-				},
-			}, true, "must be a positive duration greater than 24h"),
-		Entry("empty retentionPeriod",
-			&apisgcp.BackupBucketConfig{
-				Immutability: &apisgcp.ImmutableConfig{
-					RetentionType:   "bucket",
-					RetentionPeriod: metav1.Duration{},
-				},
-			}, true, "must be a positive duration greater than 24h"),
-		Entry("retentionPeriod less than 24 hours",
-			&apisgcp.BackupBucketConfig{
-				Immutability: &apisgcp.ImmutableConfig{
-					RetentionType:   "bucket",
-					RetentionPeriod: metav1.Duration{Duration: 23 * time.Hour},
-				},
-			}, true, "must be a positive duration greater than 24h"),
-	)
+			errs := ValidateBackupBucketCredentialsRef(credsRef, fldPath)
+			Expect(errs).To(BeEmpty())
+		})
+	})
 })
+
+func generateBackupBucketConfig(retentionType string, retentionPeriod *time.Duration, locked bool, isImmutableConfigured bool) apisgcp.BackupBucketConfig {
```

Comment:
I would call this either `isImmutabilityConfigured` or `isImmutable`. I am not sure which fits better.
```suggestion
func generateBackupBucketConfig(retentionType string, retentionPeriod *time.Duration, locked bool, isImmutabilityConfigured bool) apisgcp.BackupBucketConfig {
```